### PR TITLE
Record default array iterator functions on the library object

### PR DESCRIPTION
**Pull request by meg-gupta**
Fixes OS#14525371

### CODE REVIEW

**Comment by akroshg** [comment]
it will be good to add the test case (preferable in the jshost with mock dom objects) which deletes this properties.

File: lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp [hunk 1]
```diff
@@ -338,9 +356,9 @@ namespace Js
             library->AddMember(prototype, aliasFunctionIdentifier, scriptFunction);
         }
 
-        if (functionIdentifier == PropertyIds::values)
+        if (prototype == library->arrayPrototype)
         {
-            library->arrayPrototypeDefaultValuesFunction = func;
+            RecordDefaultIteratorFunctions(functionIdentifier, scriptContext, func);
```
Comment (akroshg) [inline lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp]: what happens if the .values function deleted or overridden? do we see the updated behavior?

Comment (meg-gupta) [reply lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp]: If values function is deleted or overridden we will see updated behavior. For for-ofs we use Symbol.iterator, we will continue to work the default way for for-ofs.

File: lib/Runtime/Library/JavascriptLibrary.cpp [hunk 2]
```diff
@@ -1674,7 +1674,7 @@ namespace Js
             }
             else
             {
-                AssertMsg(0, "arrayPrototypeKeysFunction should not be nullptr");
+                this->EnsureBuiltInEngineIsReady();
             }
```
Comment (meg-gupta) [inline lib/Runtime/Library/JavascriptLibrary.cpp]: @akroshg Updated as discussed.

**Comment by meg-gupta** [comment]
@akroshg I have added unittest with mock dom objects and sent out a PR. Is this okay to merge ?

**Pull request merged**